symforce/codegen/backends/cpp/cpp_code_printer.py

```diff
@@ -16,13 +16,31 @@ class CppCodePrinter(CXX11CodePrinter):
     behavior for codegen compatibility and efficiency.
     """
 
-    def __init__(self, settings: T.Dict[str, T.Any] = None) -> None:
+    def __init__(
+        self,
+        settings: T.Optional[T.Dict[str, T.Any]] = None,
+        override_methods: T.Optional[T.Dict[sympy.Function, str]] = None,
+    ) -> None:
         settings = dict(
             settings or {},
             math_macros={key: f"Scalar({macro})" for key, macro in get_math_macros().items()},
         )
         super().__init__(settings)
 
+        self.override_methods = override_methods or {}
+        for (expr, name) in self.override_methods.items():
+            self._set_override_methods(expr, name)
+
+    @classmethod
+    def _set_override_methods(cls, expr: sympy.Function, name: str) -> None:
+        method_name = f"_print_{str(expr)}"
+
+        def _print_expr(cls: CppCodePrinter, expr: sympy.Expr) -> str:
+            expr_string = ",".join(map(cls._print, expr.args))
```

Comment:
Probably use `", "` so it looks nice(ish) even without autoformat.  Also should be `self` instead of `cls` here I think.  I think this sort of dynamic overriding thing is fine - it might be more pythonic or whatever to have the user define a subclass but this is waaaaaaaaaaaay nicer of an interface so I'm on board

Reply:
Fair enough. Agreed with the second point, way too much boilerplate otherwise